lib/internal/modules/esm/loader.js

```diff
@@ -513,19 +519,44 @@ function getHooksProxy() {
 /**
  * Register a single loader programmatically.
  * @param {string} specifier
- * @param {string} [parentURL]
- * @returns {void}
+ * @param {string} [parentURL] Base to use when resolving `specifier`; optional if
+ * `specifier` is absolute. Same as `options.parentUrl`, just inline
+ * @param {object} [options] Additional options to apply, described below.
+ * @param {string} [options.parentURL] Base to use when resolving `specifier`
+ * @param {any} [options.data] Arbitrary data passed to the loader's `initialize` hook
+ * @param {any[]} [options.transferList] Objects in `data` that are changing ownership
+ * @returns {any} The result of the loader's initialize hook, if any
  * @example
  * ```js
  * register('./myLoader.js');
+ * register('ts-node/esm', { parentURL: import.meta.url });
+ * register('./myLoader.js', { parentURL: import.meta.url });
  * register('ts-node/esm', import.meta.url);
  * register('./myLoader.js', import.meta.url);
  * register(new URL('./myLoader.js', import.meta.url));
+ * register('./myLoader.js', {
+ *   parentURL: import.meta.url,
+ *   data: { banana: 'tasty' },
+ * });
+ * register('./myLoader.js', {
+ *   parentURL: import.meta.url,
+ *   data: someArrayBuffer,
+ *   transferList: [someArrayBuffer],
+ * });
  * ```
  */
-function register(specifier, parentURL = 'data:') {
+function register(specifier, parentURL, options) {
```

Comment:
We need to mark the parameter as optional, otherwise `register.length` will report the wrong number of mandatory arguments.
```suggestion
function register(specifier, parentURL = 'data:', options) {
```
or
```suggestion
function register(specifier, parentURL = undefined, options) {
```